buildSrc/src/main/groovy/org/elasticsearch/gradle/NoticeTask.groovy

```diff
@@ -50,46 +55,122 @@ public class NoticeTask extends DefaultTask {
     }
 
     /** Add notices from the specified directory. */
-    public void licensesDir(File licensesDir) {
+    void licensesDir(File licensesDir) {
         licensesDirs.add(licensesDir)
     }
 
+    void source(Object source) {
+        if (sources == null) {
+            sources = project.fileTree(source)
+        } else {
+            sources += project.fileTree(source)
+        }
+    }
+
+    void source(SourceDirectorySet source) {
+        if (sources == null) {
+            sources = source
+        } else {
+            sources += source
+        }
+    }
+
     @TaskAction
-    public void generateNotice() {
+    void generateNotice() {
         StringBuilder output = new StringBuilder()
         output.append(inputFile.getText('UTF-8'))
         output.append('\n\n')
         // This is a map rather than a set so that the sort order is the 3rd
         // party component names, unaffected by the full path to the various files
         Map<String, File> seen = new TreeMap<>()
-        for (File licensesDir : licensesDirs) {
-            licensesDir.eachFileMatch({ it ==~ /.*-NOTICE\.txt/ }) { File file ->
-                String name = file.name.substring(0, file.name.length() - '-NOTICE.txt'.length())
-                if (seen.containsKey(name)) {
-                    File prevFile = seen.get(name)
-                    if (prevFile.text != file.text) {
-                        throw new RuntimeException("Two different notices exist for dependency '" +
-                                name + "': " + prevFile + " and " + file)
-                    }
-                } else {
-                    seen.put(name, file)
+        noticeFiles.each { File file ->
+            String name = file.name.substring(0, file.name.length() - '-NOTICE.txt'.length())
```

Comment:
No big deal, but might this be a little easier to grok as follows?

```suggestion
            String name = file.name.replaceFirst("-NOTICE\.txt$", "")
```

Reply:
No doubt 😄